src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -122,16 +119,55 @@
    * @return completable future which might hold {@link InvoiceLine} on success or an exception if any issue happens
    */
   CompletableFuture<InvoiceLine> createInvoiceLine(InvoiceLine invoiceLine, Invoice invoice) {
-    invoiceLine.setId(UUID.randomUUID().toString());
+    invoiceLine.setId(UUID.randomUUID()
+      .toString());
     invoiceLine.setInvoiceId(invoice.getId());
 
     JsonObject line = mapFrom(invoiceLine);
 
-    return generateLineNumber(invoice)
-      .thenAccept(lineNumber -> line.put(INVOICE_LINE_NUMBER, lineNumber))
+    return generateLineNumber(invoice).thenAccept(lineNumber -> line.put(INVOICE_LINE_NUMBER, lineNumber))
+      .thenAccept(t -> calculateInvoiceLineTotals(invoiceLine, invoice))
       .thenCompose(v -> createInvoiceLineSummary(invoiceLine, line));
   }
 
+  public CompletableFuture<InvoiceLine> calculateInvoiceLineTotals(InvoiceLine invoiceLine) {
+    return getInvoice(invoiceLine).thenApply(invoice -> calculateInvoiceLineTotals(invoiceLine, invoice));
+  }
+
+  private InvoiceLine calculateInvoiceLineTotals(InvoiceLine invoiceLine, Invoice invoice) {
+    String currency = invoice.getCurrency();
+    CurrencyUnit currencyUnit = Monetary.getCurrency(currency);
+    MonetaryAmount subTotal = Money.of(invoiceLine.getSubTotal(), currencyUnit);
+
+    MonetaryAmount adjustmentTotals = calculateAdjustmentsTotal(invoiceLine, subTotal);
+    MonetaryAmount total = adjustmentTotals.add(subTotal).with(MonetaryOperators.rounding());
+    invoiceLine.setAdjustmentsTotal(adjustmentTotals.getNumber()
+      .doubleValue());
+    invoiceLine.setTotal(total.getNumber()
+      .doubleValue());
+
+    return invoiceLine;
+  }
+
+
+  private MonetaryAmount calculateAdjustmentsTotal(InvoiceLine invoiceLine, MonetaryAmount subTotal) {
+    return invoiceLine.getAdjustments()
+      .stream()
+      .map(adj -> calculateAdjustment(adj, subTotal))
+      .collect(MonetaryFunctions.summarizingMonetary(subTotal.getCurrency()))
+      .getSum()
+      .with(MonetaryOperators.rounding());
+  }
+
+
+  private MonetaryAmount calculateAdjustment(Adjustment adjustment, MonetaryAmount subTotal) {
```

Comment:
@craigmcnally @KVupp  Just one thing I would like to discuss. There is `relationToTotal` property and according to [Acquisitions Interface Fields - Invoice](https://docs.google.com/spreadsheets/d/1F88PoSnEKD7oFnza1ZuOgzAgKnK95XOyTQzxyww3tZA/edit#gid=131223376&range=A67) 
> In addition to: added to the subtotal, Included in: reported as a portion of the subtotal, Separate from: Calculated based on the subtotal and reported, but not included in the invoice total.

How this property should affect totals?

Reply:
That description is somewhat misleading...  

- Adjustments that are "In addition to" the total will affect the total.  
- "Included in" will not affect the total - This cost is baked into the product price - we want to call it out to be explicit about the breakdown of costs but don't need to add the amount to the total.
- "Separate from" will also not affect the total.  - We're going to pay this amount (e.g. maybe to some third party) but it's not actually part of what we're paying the vendor..,

Reply:
So basically "Included in" and "Separate from" are solely for bookkeeping.  "In addition" is the only one we need to worry about when calculating the total.

Reply:
so should we add the logic into the calculation now, or do we have a separate story for it in the backlog?

Reply:
it's up to you... I'm fine with either way

Reply:
ok if a story is not already present, I am handling it in this story